### DISCUSSION TITLE
Basic node input question

### DISCUSSION THREAD

**Discussion by meaningfromdata** [kedro-org/kedro#1279] [Idea]
Basic question:  Do all node inputs have to reference an element that is already listed in the catalog.yml or parameters.yml files?  In other words, is not possible to simply add an input directly in the pipeline.py file?  For example, if I want to pass a csv file and a string as the inputs to a node, how do I pass the string (note that the csv file is already in the catalog.yml)?  Does the string need to be listed in the parameters.yml file or can I define it directly in the pipeline node inputs?

**Comment by datajoely** [comment]
So you can use something called [partial application](https://en.wikipedia.org/wiki/Partial_application#:~:text=In%20computer%20science%2C%20partial%20application,producing%20a%20function%20of%20type%20.) to bind values to a function before calling them.

Here is a StackOverflow [post on how to do this](https://stackoverflow.com/questions/58875820/how-to-pass-a-literal-value-to-a-node). 

Personally I would like to introduce a Kedro `literal()` method that would make this easier for users, so if people would like to see this in the library - let us know!

**Reply by meaningfromdata** [reply]
Thanks.  I vote for including a literal() method to allow for this.  Probably judging too soon, but at a glance, partial seems like an obscure/hacky way to do this.

**Reply by meaningfromdata** [reply]
Related basic question:  Can parameters.yml have nested dictionaries?  If so, how do I pass the values from them?  For example, if I have the following in my parameters.yml file:

```
df_names:
   cohort: cohort_df
   addresses: address_df
```
How do I add df_names['addresses'] as an input to a node?  What does the string look like?
'params: df_names: addresses' and 'params: df_names['addresses']' aren't working.

**Reply by datajoely** [reply]
Yes - absolutely the syntax uses `.` dot syntax. so `df_names.addresses`

**Reply by meaningfromdata** [reply]
Ok, that fixed it.  Thanks!
